indico/modules/events/management/controllers/settings.py

```diff
@@ -139,6 +142,19 @@ class RHEditEventClassification(RHEditEventDataBase):
     form_class = EventClassificationForm
     section_name = 'classification'
 
+    def _process(self):
+        allowed_keywords = misc_settings.get('allowed_keywords')
+        if allowed_keywords:
+            default = list(filter(lambda kw: kw in allowed_keywords, self.event.keywords))
+            self.form_class.keywords = IndicoSelectMultipleCheckboxField(_('Keywords'), choices=allowed_keywords,
```

Comment:
You're modifying the form class globally. That's a bad idea - you need to subclass it and modify the subclass.

I'd implement it as a property (untested):

```python
@property
def form_class(self):
    keywords = ...  # IndicoSelectMultipleCheckboxField(...) or IndicoTagListField(...)
    return type('_EventClassificationForm', (EventClassificationForm,), {'keywords': keywords})
```

Reply:
Yes, this is a much better solution!